aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,199 @@
+  * Let developers understand and work on both client-side and server-side code.
+* Reusable, sharable UI components: Blazor apps are composed of reusable web UI components implemented using C#, HTML, and CSS. Both client and server code are written in C#, allowing you to share code and libraries.
+* Blazor works in all modern web browsers, including mobile browsers. Blazor uses open web standards without plug-ins or code transpilation.
+* Quickly get productive with Blazor reusable UI components from top component vendors like:
+  * [Telerik](https://www.telerik.com/blazor-ui)
+  * [DevExpress](https://www.devexpress.com/blazor/)
+  * [Syncfusion](https://www.syncfusion.com/blazor-components)
+  * [Radzen](https://blazor.radzen.com/)
+  * [Infragistics](https://www.infragistics.com/products/ignite-ui-blazor)
+  * [GrapeCity](https://www.grapecity.com/componentone/blazor-ui-controls)
+  * [jQWidgets](https://www.htmlelements.com/blazor/), and others.
+
+To get started with your first ASP.NET Core Blazor WebAssembly app, see [Tooling for ASP.NET Core Blazor](xref:blazor/tooling). For a complete overview of ASP.NET Core Blazor WebAssembly, its architecture and benefits, see [Overview of Blazor WebAssembly](xref:blazor/index#blazor-webassembly).
+
+### ASP.NET Core Blazor Server
+
+ASP.NET Core Blazor Server has the advantage of direct access to server resources, while providing model for building rich, interactive, and composable user interfaces in C#. Blazor Server components are UI elements such as a page, dialog, or data entry form, that handle client-side web UI interactions. The client-side web UI interactions are handled over a websocket connection, and can span multiple connections to increase speed. Blazor components keep track of the state of the client-side DOM and efficiently apply updates to only the components that need it.
+
+Blazor Server benefits:
+
+* Create rich interactive UIs using [C#](/dotnet/csharp/) instead of [JavaScript](https://www.javascript.com).
+* Share server-side and client-side app logic written in .NET.
+* Render the UI as HTML and CSS for wide browser support, including mobile browsers.
+* Share app logic across server and client.
+* Leverage the existing .NET ecosystem of [.NET libraries](/dotnet/standard/class-libraries).
+* Flexibility of access to server resources, in a protected environment. For example:
+  * Database access.
+  * Access to secrets, such as an access key used for API calls to Azure storage.
+* Uses a token-based security model, rather than using cookies that may or may not be enabled on the browser.
+
+To get started with your first ASP.NET Core Blazor Server app, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For a complete overview of ASP.NET Core Blazor Server, its architecture and benefits, see [Introduction to ASP.NET Core Blazor](xref:blazor/index).
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
```

Comment:
We should generalize this section a bit and link to existing content. You can use Blazor components with both Razor Pages and Views. Those components can be hosted using Blazor Server or Blazor WebAssembly. The components can be prerendered or not. This is covered in detail here: https://docs.microsoft.com/aspnet/core/blazor/components/prerendering-and-integration (note the hosting model pivot in the doc).  In this doc we just need to make some general statements about how Blazor components can be rendered or even prerendered from views and pages. Maybe this should just be listed as a benefit of using Blazor instead of being an entire section?

Reply:
@danroth27, I could re-organizing both Blazor Server and Blazor WebAssembly to be a subset of Blazor which quickly introduces Blazor Components and benefits first.    
Here is what it would look like:
- Blazor (Section heading)
-- Blazor components and their benefits plus flexibility of being used for Razor Pages and Views, plus the flexibility of being used in both Blazor Server and Blazor WebAssembly hosting options.
- - -Blazor Server quick intro, benefits, links.
- - -Blazor WebAssembly quick intro, benefits, links.

Would that work better?  Or leave the org as is with Blazor Server and Blazor Web Assembly (but of course fix description of Blazor components as you mentioned above in either case.)

Reply:
Updated to the organization I just pointed out above.  See what you think.